docs/proposals/runtime-client-server.md

```diff
@@ -0,0 +1,183 @@
+- Manage api interfaces simply: server and client interfaces are generated automatically
+
+A new container runtime manager `KubeRuntimeManager` will be introduced to kubelet, which will
+
+- conforms to kubelet's [Runtime](../../pkg/kubelet/container/runtime.go#L58) interface
+- manage Pods and Containers lifecycle according to kubelet policies
+- call remote runtime's API to perform specific pod, container or image operations
+
+A simple workflow of invoking remote runtime API on starting a Pod can be shown:
+
+```
+Kubelet                  KubeRuntimeManager                RemoteRuntime
+   +                              +                               +
+   |                              |                               |
+   +---------SyncPod------------->+                               |
+   |                              |                               |
+   |                              +--SandboxManager.Create()----->+
+   |                              |                               |
+   |                              XXXXXXXXXXXX                    |
+   |                              |          X                    |
+   |                              |    NetworkPlugin.             |
+   |                              |       SetupPod                |
+   |                              |          X                    |
+   |                              XXXXXXXXXXXX                    |
+   |                              |                               |
+   |                              +<------------------------------+
+   |                              +--ImageManager.Pull()--------->+
+   |                              +<------------------------------+
+   |                              +--ContainerManager.Create()--->+
+   |                              +<------------------------------+
+   |                              +--ContainerManager.Start()---->+
+   |                              +<------------------------------+
+   |                              |                               |
+   | <-------Success--------------+                               |
+   |                              |                               |
+   +                              +                               +
+```
+
+**API definition**
+
+Since we are going to introduce more image formats and want to separate image management from containers and pods, this proposal introduces two services `RuntimeService` and `ImageService`. Both services are defined at [pkg/kubelet/api/v1alpha1/runtime/api.proto](../../pkg/kubelet/api/v1alpha1/runtime/api.proto):
+
+```proto
+// Runtime service defines the public APIs for remote container runtimes
+service RuntimeService {
+    // Version returns the runtime name, runtime version and runtime API version
+    rpc Version(VersionRequest) returns (VersionResponse) {}
+    // Status returns error if the runtime is unhealthy; nil otherwise
+    rpc Status(StatusRequest) returns (StatusResponse) {}
```

Comment:
@vishh Sorry, I forgot to remove this interface from doc (it is already removed from api.proto). As @yujuhong suggested, kubelet will use `Version()` api for checking the runtime's health.